### DISCUSSION TITLE
Sentences repeating in text

### DISCUSSION THREAD

**Discussion by akanervaa** [openai/whisper#1059] [Q&A] [answered]
I am having an issue where sentences are repeating themselves on the result of transcribe function. In terminal output there is no repetition so I'm wondering what could be the issue. I use cuda 2.1 and torch is newest version+cu117. Tried other versions of torch, same issue. I have tried changing some parameters such as previous audio and compression smthing they dont do anything. My friend with same code has no issues.

**Comment by jongwook** [comment, marked as answer]
Hi, I made a fix for this a few hours ago; can you try again with the latest commit, or 

```
pip install openai-whisper==20230308
```

?

**Reply by akanervaa** [reply]
Works like a charm now, respect for quick fix.

**Reply by parkuman** [reply]
This also fixed the issue for me. I thought it was related to #977 but thankfully this fixed it! Thank you for the quick fix.

**Reply by jongwook** [reply]
glad it worked!

**Reply by ScriptTiger** [reply]
I just ran into this issue and temporarily reverted to an older version. But super glad this was addressed so quickly! Thanks @akanervaa and @jongwook!!! I'll definitely try using the latest version next time.

**Reply by thiwa8** [reply]
You could check the validity and in doubt have a second go with a different transcription function.

`def is_transcription_valid(text: str) -> bool:
    """Validates transcription quality"""
    if not text:
        logger.warning("Empty transcription.")
        return False

    words = [w.lower() for w in text.split()]
    
    # Check for three sequential repetitions
    for i in range(len(words) - 2):
        if words[i] == words[i+1] == words[i+2]:
            logger.warning(f"Sequential repetition detected: '{words[i]}' repeated 3 times")
            return False

    # Check for repeated n-grams (bi/tri/quad)
    for n in [2, 3, 4]:  # Check bi-grams, tri-grams, and quad-grams
        for i in range(len(words) - (n * 2 - 1)):
            ngram = tuple(words[i:i+n])
            next_ngram = tuple(words[i+n:i+n*2])
            if ngram == next_ngram:
                logger.warning(f"Repeated {n}-gram detected: '{' '.join(ngram)}'")
                return False

    return True`

**Comment by projects-g** [comment]
I installed the latest available whisper codebase but still got the same error. Some particular sets of words keep repeating over and over while transcribing. Has the issue resurfaced?

WHISPER_MODEL_SIZE=medium

`And the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, holding these `

**Reply by ScriptTiger** [reply]
The problem this issue was describing was literally every sentence was being repeated, basically outputting multiple predictions for everything, not just "particular sets of words." The problem you are describing has always happened from time to time. It will naturally lessen over time as the project improves, but there is no clear way to eliminate it all together in one fell swoop.

**Reply by phineas-pta** [reply]
it's the hallucinations problem, happened when there's silence / non speech segments

the technical solution is to use VAD

see #679
